SU2_CFD/src/output/filewriter/CParaviewXMLFileWriter.cpp

```diff
@@ -0,0 +1,682 @@
+
+      /*--- For now, create a temp 1D buffer to load up the data for writing.
+       This will be replaced with a derived data type most likely. ---*/
+
+      for (iPoint = 0; iPoint < myPoint; iPoint++) {
+        float val = (float)dataSorter->GetData(VarCounter,iPoint);
+        scalar_buf[iPoint] = val;
+      }
+      
+      WriteDataArray((void*)scalar_buf, VTKDatatype::FLOAT32, myPoint, GlobalPoint, nPoint_Cum[rank]);
+      
+      delete [] scalar_buf; scalar_buf = NULL;
+      VarCounter++;
+    }
+
+  }
+  
+  WriteString("</AppendedData>\n", MASTER_NODE);
+  WriteString("</VTKFile>\n", MASTER_NODE);
+  
+#ifdef HAVE_MPI
+  /*--- All ranks close the file after writing. ---*/
+
+  MPI_File_close(&fhw);
+#else
+  fclose(fhw);
+#endif
+
+  /*--- Compute and store the write time. ---*/
+
+#ifndef HAVE_MPI
+  StopTime = su2double(clock())/su2double(CLOCKS_PER_SEC);
+#else
+  StopTime = MPI_Wtime();
+#endif
+  UsedTime = StopTime-StartTime;
+
+  /*--- Communicate the total file size for the restart ---*/
+
+#ifdef HAVE_MPI
+  su2double my_file_size = file_size;
+  SU2_MPI::Allreduce(&my_file_size, &file_size, 1,
+                     MPI_DOUBLE, MPI_SUM, MPI_COMM_WORLD);
+#endif
+
+  /*--- Compute and store the bandwidth ---*/
+
+  Bandwidth = file_size/(1.0e6)/UsedTime;
+
+  /*--- Delete the offset counters that we needed for MPI IO. ---*/
+
+  delete [] nElem_Snd;        delete [] nElem_Cum;
+  delete [] nElemStorage_Snd; delete [] nElemStorage_Cum;
+  delete [] nPoint_Snd;       delete [] nPoint_Cum;
+
+}
+
+
+void CParaviewXMLFileWriter::WriteString(std::string str, int rank){
+
+#ifdef HAVE_MPI  
+  /*--- Reset the file view before writing the next ASCII line for cells. ---*/
+
+  MPI_File_set_view(fhw, 0, MPI_BYTE, MPI_BYTE,
+                    (char*)"native", MPI_INFO_NULL);
+  
+  if (SU2_MPI::GetRank() == rank)
+    MPI_File_write_at(fhw, disp, str.c_str(), strlen( str.c_str()),
+                      MPI_CHAR, MPI_STATUS_IGNORE);
+  disp += strlen( str.c_str())*sizeof(char);
+  file_size += sizeof(char)*strlen( str.c_str());
+#else
+  char str_buf[255];
+  strcpy(str_buf, str.c_str());
+  fwrite(str_buf, sizeof(char), strlen(str_buf), fhw);
+  file_size += sizeof(char)*strlen(str_buf);
+#endif
+  
+}
+
+void CParaviewXMLFileWriter::WriteDataArray(void* data, VTKDatatype type, unsigned long arraySize, 
+                                            unsigned long globalSize, unsigned long offset){
+  
+
+  unsigned long totalByteSize, byteSize;
+  
+  std::string typeStr;
+  unsigned long typeSize;
+  switch (type) {
+    case VTKDatatype::FLOAT32:
+      typeStr = "\"Float32\"";
+      typeSize = sizeof(float);
+      break;
+    case VTKDatatype::INT32:
+      typeStr = "\"Int32\"";
+      typeSize = sizeof(int);
+      break;
+    case VTKDatatype::UINT8:
+      typeStr = "\"UInt8\"";
+      typeSize = sizeof(char);     
+      break;
+    default:
+      SU2_MPI::Error("Unknown Type", CURRENT_FUNCTION);
+      break;
+  }
```

Comment:
This is done also in AddDataArray, maybe it could be in a function to make maintenance easier if more types are added?